CLI: Harden offline signing and tests

t-nelson opened the pull request:
#### Problem

Various holes exists in CLI tests for offline signing allowing operations intended to be offline to perform cluster queries

#### Summary of Changes

Create a separate `CliConfig` for offline operations, verify that it is offline and use it
Fix everything that breaks as a result

Requires #8009

t-nelson commented:
@danpaul000 have you run into any issues with these changes in your testing?

danpaul000 commented:
> @danpaul000 have you run into any issues with these changes in your testing?

Let me re-run my tests now that the #8009 is landed on top of this branch and I'll let you know ASAP.

t-nelson commented:
Awesome!  Thanks!  Logic should be nearly identical to what you've been using, just added tests and  cleaned things up.  Worth the re-test for sure though

jackcmay commented:
lgtm
